inference_segmentation.py

```diff
@@ -282,7 +283,36 @@ def calc_inference_chunk_size(gpu_devices_dict: dict, max_pix_per_mb_gpu: int =
     return max_chunk_size_rd
 
 
-def main(params: dict) -> None:
+def override_model_params_from_checkpoint(
+        params: DictConfig,
+        checkpoint_params):
+    """
+    Overrides model-architecture related parameters from provided checkpoint parameters
+    @param params: Original parameters as inputted through hydra
+    @param checkpoint_params: Checkpoint parameters as saved during checkpoint creation when training
+    @return:
+    """
+    modalities = get_key_def('modalities', params['dataset'], expected_type=Sequence)
+    classes = get_key_def('classes_dict', params['dataset'], expected_type=(dict, DictConfig))
+
+    modalities_ckpt = get_key_def('modalities', checkpoint_params['dataset'], expected_type=Sequence)
+    classes_ckpt = get_key_def('classes_dict', checkpoint_params['dataset'], expected_type=(dict, DictConfig))
+    model_ckpt = get_key_def('model', checkpoint_params, expected_type=(dict, DictConfig))
+
+    if model_ckpt != params.model or classes_ckpt != classes or modalities_ckpt != modalities:
+        logging.warning(f"\nParameters from checkpoint will override inputted parameters."
+                        f"\n\t\t\t Inputted | Overriden"
```

Comment:
If a parameter read from the hydra config is overridden on the commandline, that will still be overridden by the checkpoint right ?

Reply:
yep!